march_hardware/src/IMotionCube.cpp

```diff
@@ -16,138 +16,39 @@ IMotionCube::IMotionCube(int slaveIndex, Encoder encoder) : Slave(slaveIndex)
-  // download 1A01 pdo count: 4
-  success &= sdo_bit32(slaveIndex, 0x1A01, 0, 3);
-
-  // clear 1A02 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1A02, 0, 0);
-  // download 1A02 pdo entries
-
-  //  Torque actual value
-  success &= sdo_bit32(slaveIndex, 0x1A02, 1, 0x60770010);
-  //  Current limit
-  success &= sdo_bit32(slaveIndex, 0x1A02, 2, 0x207f0010);
-  //  Motor position
-  success &= sdo_bit32(slaveIndex, 0x1A02, 3, 0x20880020);
-
-  // download 1A02 pdo count: 1
-  success &= sdo_bit32(slaveIndex, 0x1A02, 0, 3);
-  // clear 1A03 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1A03, 0, 0);
-
-  //----------------------------------------
-
-  // clear 1600 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1600, 0, 0);
-  // download 1600 pdo entries
-  //  Control word
-  success &= sdo_bit32(slaveIndex, 0x1600, 1, 0x60400010);
-  //  Target position
-  success &= sdo_bit32(slaveIndex, 0x1600, 2, 0x607A0020);
-  // download 1600 pdo count: 2
-  success &= sdo_bit32(slaveIndex, 0x1600, 0, 2);
-
-  //--------------------
-
-  // clear 1601 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1601, 0, 0);
-  // clear 1602 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1602, 0, 0);
-  // clear 1603 pdo entries
-  success &= sdo_bit32(slaveIndex, 0x1603, 0, 0);
-
-  //----------------------------------------
-
-  // download 1C12:01 index
-  success &= sdo_bit16(slaveIndex, 0x1C12, 1, 0x1600);
-  // download 1C12 counter
-  success &= sdo_bit8(slaveIndex, 0x1C12, 0, 1);
-
-  //--------------------
-
-  // download 1C13:01 index
-  success &= sdo_bit16(slaveIndex, 0x1C13, 1, 0x1A00);
-  success &= sdo_bit16(slaveIndex, 0x1C13, 2, 0x1A01);
-  success &= sdo_bit16(slaveIndex, 0x1c13, 3, 0x1A02);
-  // download 1C13 counter
-  success &= sdo_bit8(slaveIndex, 0x1C13, 0, 3);
-
-  //----------------------------------------
-
-  if (success)
-  {
-    ROS_INFO("Successfully mapped PDOs of IMC %d", this->slaveIndex);
-  }
-  else
-  {
-    ROS_WARN("PDO mapping of IMC %d failed", this->slaveIndex);
-  }
-
-  return success;
+  PDOmap pdoMapMISO = PDOmap();
+  pdoMapMISO.addObject(IMCObjectName::StatusWord);      // Compulsory!
+  pdoMapMISO.addObject(IMCObjectName::ActualPosition);  // Compulsory!
+  pdoMapMISO.addObject(IMCObjectName::DCLinkVoltage);
+  pdoMapMISO.addObject(IMCObjectName::DetailedErrorRegister);
+  this->misoByteOffsets = pdoMapMISO.map(this->slaveIndex, dataDirection::miso);
+
+  PDOmap pdoMapMOSI = PDOmap();
+  pdoMapMOSI.addObject(IMCObjectName::ControlWord);  // Compulsory!
+  pdoMapMOSI.addObject(IMCObjectName::TargetPosition);
+  this->mosiByteOffsets = pdoMapMOSI.map(this->slaveIndex, dataDirection::mosi);
 }
 
 // Set configuration parameters to the IMC
 bool IMotionCube::writeInitialSettings(uint8 ecatCycleTime)
 {
   bool success = true;
-
+  // sdo_bit32(slaveIndex, address, subindex, value);
   // mode of operation
   success &= sdo_bit8(slaveIndex, 0x6060, 0, 8);
```

Comment:
Maybe a bit out of scope but shall we refactor this to
```
   success &= sdo_bit8(slaveIndex, MODE_OF_OPERATION_ADDRESS, 0, 8);
```
Then we no longer need the comment

Reply:
Agree, but not for this PR

Reply:
https://github.com/project-march/hardware-interface/issues/30